### PR TITLE
AC-706 disabled user screen interaction while patient is being regist…

### PR DESCRIPTION
**Pull request by MasterBlaster99**
## Description of what I changed
I have removed the user-screen interaction while the patient is being registered 

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-706

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [ ] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

 

https://user-images.githubusercontent.com/71667923/107322169-f36e6080-6ac9-11eb-91f2-a7e1f8631c44.mp4

### CODE REVIEW

**Comment by MasterBlaster99** [comment]
@rishabh-997, this only works when the progress bar is visible
it works after checking the Edit Texts

**Comment by MasterBlaster99** [comment]
> @rishabh-997,now this works fine
> this only works when the progress bar is visible
> it works after checking all the Edit Texts

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 1]
```diff
@@ -379,6 +380,7 @@ public void hideSoftKeys() {
     @Override
     public void setProgressBarVisibility(boolean visibility) {
         binding.progressBar.setVisibility(visibility ? View.VISIBLE : View.GONE);
+        getActivity().getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
```
Comment (rishabh-997) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java:383]: Can you also add a 70-80% translucent view like this which is visible during loading?
https://images.app.goo.gl/e1Rx9tGZXUuhKR6i8

**Review by rishabh-997 — CHANGES_REQUESTED** [review]
Add this too!

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java (hunk 1 above)
Comment (MasterBlaster99) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java:385]: added

**Comment by MasterBlaster99** [comment]
> Add this too!

Added the 70% translucent View @rishabh-997 please Review !!

**Comment by MasterBlaster99** [comment]
> > Add this too!
> 
> Added the 70% translucent View @rishabh-997 please Review !!

WhatsApp Image 2021-02-17 at 12 38 27 PM

**Comment by MasterBlaster99** [comment]
@rishabh-997  please review !!

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 2]
```diff
@@ -379,6 +382,13 @@ public void hideSoftKeys() {
...
+        Dialog pd = new Dialog(getActivity(), android.R.style.Theme_Black);
+        View view = LayoutInflater.from(getActivity()).inflate(R.layout.activity_patient_info, null);
+        pd.requestWindowFeature(Window.FEATURE_NO_TITLE);
+        pd.getWindow().setBackgroundDrawableResource(R.color.translucent_70_percent);
+        pd.setContentView(view);
+        pd.show();
```
Comment (rishabh-997) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java:391]: how are you countering repeated usage??? The function `setProgressBarVisibility` does both switching and hiding ProgressBar, but how is the state changed for view???

**Review by rishabh-997 — CHANGES_REQUESTED** [review]
Address this query.

**Comment by MasterBlaster99** [comment]
> Address this query.

😕 i am not getting this , what should i do exactly ?

**Comment by rishabh-997** [comment]
> > Address this query.
> 
> 😕 i am not getting this , what should i do exactly ?

suppose you cluck submit and progress bar starts rotating along with translucent background... now suppose there is some error which makes the setProgressBarVisibility function false and progress bar stops, in that case how will the translucent background and touch interaction stop???

**Comment by MasterBlaster99** [comment]
> > > Address this query.
> > 
> > 
> > 😕 i am not getting this , what should i do exactly ?
> 
> suppose you cluck submit and progress bar starts rotating along with translucent background... now suppose there is some error which makes the setProgressBarVisibility function false and progress bar stops, in that case how will the translucent background and touch interaction stop???

i have made the changes ! when an error occurs and setProgressBarVisibility is set to false, now the user will get back the screen interaction. But i have removed the translucent background , i am not able remove it if the setProgressBar is false

**Comment by rishabh-997** [comment]
> removed the translucent background

Try to find a solution for this, there is nothing that can't be done, all it takes is innovative thought and stackOverflow

**Comment by MasterBlaster99** [comment]
Done!!
@rishabh-997 please review

> > removed the translucent background
> 
> Try to find a solution for this, there is nothing that can't be done, all it takes is innovative thought and stackOverflow

Done!!
@rishabh-997 please review

**Comment by MasterBlaster99** [comment]
@rishabh-997  please review!

**Review by LuGO0 — APPROVED** [review]
LGTM @rishabh-997 we can merge this.

**Comment by MasterBlaster99** [comment]
> LGTM @rishabh-997 we can merge this.

@rishabh-997  yes 🥺

**Pull request merged**